jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/model/PackageNode.java

```diff
@@ -187,15 +222,23 @@ public static PackageNode createNodeForClasspathEntry(IClasspathEntry classpathE
         return null;
     }
 
-    public static PackageNode createNodeForTypeRoot(IJavaElement typeRoot) {
-        PackageNode typeRootNode = new TypeRootNode(typeRoot.getElementName(), typeRoot.getPath().toPortableString(), NodeKind.TYPEROOT,
-                typeRoot instanceof IClassFile ? TypeRootNode.K_BINARY : TypeRootNode.K_SOURCE);
-        if (typeRoot instanceof ICompilationUnit) {
-            typeRootNode.setUri(JDTUtils.toURI((ICompilationUnit) typeRoot));
-        } else if (typeRoot instanceof IClassFile) {
-            typeRootNode.setUri(JDTUtils.toUri((IClassFile) typeRoot));
+    public static PackageNode createNodeForPrimaryType(IType type) {
+        PackageNode primaryTypeNode = new PackageNode(type.getElementName(), type.getPath().toPortableString(), NodeKind.PRIMARYTYPE);
```

Comment:
why create packageNode for IType?

Reply:
That's some unknown historical reason. The Java side code only has the following kinds of nodes:
- PackageNode
- ContainerNode
- PackageRootNode